lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -177,6 +189,192 @@ impl EnsoLexer {
+            }
+        }
+    }
+
+    /// Triggered when ending a doc comment.
+    fn on_end_doc_comment<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Comment::on_end_doc_comment");
+        let doc_comment = self.doc_comment;
+        let mut comment = self.comment_state.consume_current();
+        let blank_lines = comment.consume_blank_lines();
+        if !blank_lines.is_empty() {
+            self.block_state.seen_newline = true;
+        }
+        self.block_state.delayed_append_lines.extend(blank_lines);
+        let comment_token = comment.into();
+        self.append_token(comment_token);
+        self.pop_states_including(doc_comment);
+    }
+
+    /// Triggered when a new line is discovered in a doc comment.
+    fn doc_comment_on_new_line<R:LazyReader>(&mut self, reader:&mut R) {
+        trace!(self.logger,"Comment::doc_comment_on_new_line");
+        self.pop_state();
+        let indent         = self.consume_current().chars().count();
+        let comment_indent = self.comment_state.current_comment.indent;
+        if indent < comment_indent {
+            self.on_end_doc_comment(reader);
+            self.block_on_newline(token::LineEnding::None,reader);
+        } else if (indent - comment_indent) > 0 {
+            let remaining_indent = " ".repeat(indent - comment_indent);
+            self.comment_state.append_to_line(remaining_indent);
+        }
+    }
+
+    /// Triggered when an empty line is discovered in a doc comment.
+    fn doc_comment_on_empty_line<R:LazyReader>
+    ( &mut self
+    , line_ending:token::LineEnding
+    , reader:&mut R
+    ) {
+        trace!(self.logger,"Comment::doc_comment_on_empty_line");
+        let current = self.consume_current();
+        let indent  = current.chars().count() - line_ending.size();
+        self.comment_state.submit_blank_line(indent,line_ending);
+        if let token::LineEnding::None = line_ending {
+            let block_newline = self.block_newline;
+            self.on_end_doc_comment(reader);
+            self.push_state(block_newline);
+            self.on_eof(reader);
+        }
+    }
+
+    // === Comment Rules ===
+
+    /// The rules for lexing Enso comments.
+    fn add_comment_rules(lexer:&mut EnsoLexer) {
+        let comment_char      = c!('#');
+        let lf                = l!("\n");
+        let crlf              = l!("\r\n");
+        let eof               = Pattern::eof();
+        let spaces            = EnsoLexer::spaces();
+        let doc_comment_start = &comment_char >> &comment_char >> &spaces;
+        let eof_newline       = &spaces.opt() >> &eof;
+        let empty_line_lf     = &spaces.opt() >> &lf;
+        let empty_line_crlf   = &spaces.opt() >> &crlf;
+
+        let initial_state_id = lexer.initial_state;
+        let initial_state    = lexer.group_mut(initial_state_id);
+        initial_state.create_rule(&doc_comment_start,"self.on_begin_doc_comment(reader)");
```

Comment:
make macro for it, which could look like:

```
RULE! { initial_state(doc_comment_start) => self.on_begin_doc_comment(reader)}
```

It will be much easier to read that this is something special and we will nto have so much rust code in strings anymore :)

Reply:
Use stringify.